monai/transforms/compose.py

```diff
@@ -73,16 +74,19 @@ def evaluate_with_overrides(
         override: keyword arguments to apply transforms.
         override_keys: to which the override arguments are used when apply transforms.
         verbose: whether to print debugging info when evaluate MetaTensor with pending operations.
+        mode: "default" or "strict".
 
     """
     if not lazy_evaluation:
         return data  # eager evaluation
     overrides = (overrides or {}).copy()
     if isinstance(data, monai.data.MetaTensor):
+        if isinstance(upcoming, mt.MapTransform) and override_keys not in upcoming.keys:
+            return data
         if data.has_pending_operations and (
             (upcoming is None)
-            or (isinstance(upcoming, mt.Identity))
-            or (isinstance(upcoming, mt.Identityd) and override_keys in upcoming.keys)
+            or (isinstance(upcoming, (mt.Identity, mt.Identityd)))
```

Comment:
Identity and Identityd should not be used as a synchronisation mechanism. There should be proper transforms that have the specific responsibility of causing this to occur.

Reply:
Hi all, as agreed on Friday on the team meeting, we should be working on merging Ben's codebase and not on changing the previous implementation.

Reply:
Hi @mjorgecardoso, we agreed to have a week's time. I've been actively helping Ben review the design and providing advice, unfortunately, the codebase still has large technical gaps as I commented on the PR. On the other hand, we don't find any issue on the current dev branch.